bundles/org.eclipse.e4.ui.workbench.addons.swt/src/org/eclipse/e4/ui/workbench/addons/cleanupaddon/CleanupAddon.java

```diff
@@ -381,4 +384,32 @@ private void subscribeRenderingChanged(
 	boolean isLastEditorStack(MUIElement element) {
 		return modelService.isLastEditorStack(element);
 	}
+
+	private void transferPrimaryDataStackIfRemoved(MElementContainer<?> removedElement) {
+		if (!PRIMARY_DATA_STACK_ID.equals(removedElement.getElementId())) {
+			return;
+		}
+		MPartStack newPrimaryDataStack = findFirstPartStackToBeRendered(removedElement.getParent());
+		if (newPrimaryDataStack != null) {
+			newPrimaryDataStack.setElementId(removedElement.getElementId());
```

Comment:
I think it's slightly more obvious that you assign this as the PRIMARY_DATA_STACK_ID if you use that constant rather than call a method to return what you established above must be the value of that ID.

Reply:
Makes sense. I have now completely moved the functionality for making a "primary data stack" out of a part stack and for identifying an element as such to a utility, which is used at all places that replicated the according functionality.